Update `desktop-themes/GreenLaguna/window-controls.css`: 1) Fixes the "Restore Down" button in Mozilla Firefox and Thunderbird 2) Fixes the "Close" button in dconf-editor when compositing-manager is off

davidhedlund opened the pull request:
# To create GreenLaguna-Test it in Ubuntu MATE
```
#!/bin/bash

SRC="GreenLaguna"
SUFFIX="-Test"
DST="${SRC}${SUFFIX}"
THEMES_DIR="$HOME/.themes"

# Copy theme
cp -r /usr/share/themes/"$SRC"/ "$THEMES_DIR/$DST"

# Modify index.theme
sed -i -e "s/^Name=$SRC$/Name=$DST/" -e "s/^GtkTheme=$SRC$/GtkTheme=$DST/" "$THEMES_DIR/$DST/index.theme"

# Update window-controls.css
wget -O /home/x/.themes/GreenLaguna-Test/gtk-3.0/window-controls.css \
https://raw.githubusercontent.com/mate-desktop/mate-themes/75555f4234b0d3795c63d0ba3700af3ab077daf3/desktop-themes/GreenLaguna/gtk-3.0/window-controls.css
```

# Fixes

## For all distros: Fixed the "Restore Down" button in Mozilla Firefox and Thunderbird
Fixed the GreenLaguna theme:

- Preserved the last 13 lines in [GreenLaguna/gtk-3.0/window-controls.css](https://github.com/mate-desktop/mate-themes/blob/master/desktop-themes/GreenLaguna/gtk-3.0/window-controls.css)
- Replaced the remainder of that file with the contents from [BlackMATE/gtk-3.0/window-controls.css](https://github.com/mate-desktop/mate-themes/blob/master/desktop-themes/BlackMATE/gtk-3.0/window-controls.css)

This resolves the issue in the GreenLaguna variant of the BlueMenta/Menta/GreenLaguna themes, where the "Restore Down" button appeared blank in Mozilla Firefox and Thunderbird (see [mate-desktop/mate-themes#314](https://github.com/mate-desktop/mate-themes/issues/314)).

## For Ubuntu MATE 22.04, 23.04, and 24.04: Fixed the "Close" button in dconf-editor when compositing-manager is off

1. Set the theme to original GreenLaguna
2. Run `gsettings set org.mate.Marco.general compositing-manager false`
3. To make sure that the composition-manager setting take effect on the applied theme (this is important, otherwise the close button will not disappear!):
  - 1. Close dconf-editor
  - 2. Start dconf-editor, and observe that the close button now has disappeared:
4. Set the theme to a modified GreenLaguna theme with this PR and observe that the button is now visible.

See https://gitlab.trisquel.org/trisquel/package-helpers/-/issues/169 for more information

# Note
This solution could not be applied to BlueMenta or Menta, as it would remove all window control buttons in those themes.

davidhedlund committed:
Update window-controls.css

Fixed the GreenLaguna theme:

- Preserved the last 13 lines in `GreenLaguna/gtk-3.0/window-controls.css`
- Replaced the remainder of that file with the contents from `BlackMATE/gtk-3.0/window-controls.css`

This resolves the issue in the GreenLaguna variant of the BlueMenta/Menta/GreenLaguna themes, where the "Restore Down" button appeared blank in Mozilla Firefox and Thunderbird (see [mate-desktop/mate-themes#314](https://github.com/mate-desktop/mate-themes/issues/314)).

**Note:**  
This solution could not be applied to BlueMenta or Menta, as it would remove all window control buttons in those themes.

lukefromdc reviewed (approved):
I just confirmed both the issue and that this fix works in Librewolf, which is a privacy-minded Firefox fork. We now get the middle button working like the others.

lukefromdc commented:
Calling for a second review as this works but probably changes out more code than is needed to fix one button

lukefromdc reviewed (commented):
This definately works, but probably changes out far more code than is needed to fix one button failing to render. I've done this myself as part of developing code, but always tried to drill down into why it works to minimize code churn.

On the other hand, this is a significant bug and needs to be fixed and this does work. Asking others what they think here

davidhedlund committed:
Update window-controls.css

Used the GitHub files instead of the local files from my distro; an extra `	padding: 5px;` was found in https://github.com/mate-desktop/mate-themes/blob/master/desktop-themes/BlackMATE/gtk-3.0/window-controls.css

davidhedlund commented:
@lukefromdc

With the original GreenLaguna theme, was the close button visible for you in dconf-editor? On my system, it wasn’t visible, but my patch fixed it!

So my PR solves both:
* Close button in dconf-editor
* Restore Down button in Firefox and Thunderbird

You can find more details here: https://gitlab.trisquel.org/trisquel/package-helpers/-/issues/169

lukefromdc commented:
> @lukefromdc
> 
> With the original GreenLaguna theme, was the close button visible for you in dconf-editor? On my system, it wasn’t visible, but my patch fixed it!
> 
> So my PR solves both:
> 
>     * Close button in dconf-editor
> 
>     * Restore Down button in Firefox and Thunderbird
> 
> 
> You can find more details here: https://gitlab.trisquel.org/trisquel/package-helpers/-/issues/169

Never looked for it, I really don't know as the only time I am ever running a theme other than my own UbuntuStudio-Legacy theme is when I am testing another theme. In a quick test in my normal mate-wayland session I still saw it, but I don't remember if this behaves differently in x11 and I do not have time right now to close out my whole session and start over.

Note that we may have different versions of dconf-editor

davidhedlund commented:
> > @lukefromdc
> > With the original GreenLaguna theme, was the close button visible for you in dconf-editor? On my system, it wasn’t visible, but my patch fixed it!
> > So my PR solves both:
> > ```
> > * Close button in dconf-editor
> > 
> > * Restore Down button in Firefox and Thunderbird
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > You can find more details here: https://gitlab.trisquel.org/trisquel/package-helpers/-/issues/169
> 
> Never looked for it, I really don't know as the only time I am ever running a theme other than my own UbuntuStudio-Legacy theme is when I am testing another theme. In a quick test in my normal mate-wayland session I still saw it, but I don't remember if this behaves differently in x11 and I do not have time right now to close out my whole session and start over.
> 
> Note that we may have different versions of dconf-editor

The issue applies to both X11 (Trisquel GNU/Linux) and Wayland (Ubuntu MATE).

I added this to the top post:

> # For Ubuntu MATE 22.04, 23.04, and 24.04: Fixed the "Close" button in dconf-editor when compositing-manager is off
> 
> 1. Set the theme to original GreenLaguna
> 2. Run `gsettings set org.mate.Marco.general compositing-manager false`
> 3. To make sure that the composition-manager setting take effect on the applied theme (this is important, otherwise the close button will not disappear!):
>   - 1. Close dconf-editor
>   - 2. Start dconf-editor, and observe that the close button now has disappeared:
> 4. Set the theme to a modified GreenLaguna theme with this PR and observe that the button is now visible.
> 
> See https://gitlab.trisquel.org/trisquel/package-helpers/-/issues/169 for more information

davidhedlund commented:
I wrote a Bash script in the top post to make it easy to generate GreenLaguna-Test by utilizing `wget -O /home/x/.themes/GreenLaguna-Test/gtk-3.0/window-controls.css \
https://raw.githubusercontent.com/mate-desktop/mate-themes/75555f4234b0d3795c63d0ba3700af3ab077daf3/desktop-themes/GreenLaguna/gtk-3.0/window-controls.css`

lukefromdc commented:
I think I still have the MODIFIED version (this PR) installed and that's why I didn't see the issue in dconf-editor

davidhedlund commented:
Hi @vkareh, thank you for all your help. My PR has been waiting for review for four months. Could you please take a look when you have a chance? Let me know if you need any more info.

vkareh reviewed (approved):
This works well for me, and now the window-control.css file for BlackMATE and GreenLaguna theme are a lot more similar to one another. Good change :+1:

vkareh committed:
Merge branch 'master' into master

davidhedlund commented:
> This works well for me, and now the window-control.css file for BlackMATE and GreenLaguna theme are a lot more similar to one another. Good change 👍

Agreed, that makes the themes a little bit closer to a seamless standard. Thank you so much for everything!
